src/ml/neural_net/style_transfer/mps_style_transfer_backend.mm

```diff
@@ -0,0 +1,145 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+#import <ml/neural_net/style_transfer/mps_style_transfer_backend.hpp>
+#import <ml/neural_net/style_transfer/mps_style_transfer.h>
+#import <ml/neural_net/mps_utils.h>
+
+@interface TCMPSStyleTransferHelpers:NSObject
++(float_array_map) fromNSDictionary: (NSDictionary<NSString *, NSData *> *) dictionary;
++(NSDictionary<NSString *, NSData *> *) toNSDictionary: (float_array_map) map;
+@end
+
+@implementation TCMPSStyleTransferHelpers
++(float_array_map) fromNSDictionary: (NSDictionary<NSString *, NSData *> *) dictionary {
+  float_array_map map;
+  for (NSString* key in dictionary) {
+    NSData* value = [dictionary objectForKey:key];
+
+    size_t dataLength = (size_t) (value.length / sizeof(float));
+    std::vector<size_t> dataShape = { dataLength }; 
+    float *dataBytes = (float *) (value.bytes);
+    
+    shared_float_array array = shared_float_array::copy(dataBytes, dataShape);
```

Comment:
Could you leave a TODO here to optimize away this copy? Also worth filing a GH issue for the convergence sprint. This can instead be a wrapper around the NSData: a custom subclass of `float_array` that preserves a strong reference to the NSData instance